### PR TITLE
Update references to amphtml/builtins

### PR DESCRIPTION
**Pull request by rcebulko**
Companion PR to https://github.com/ampproject/amphtml/pull/35139, which move `builtins/` to `src/builtins/`.
This PR updates documentation links to the `builtins/` folder.

Context: https://github.com/ampproject/amphtml/issues/34866

### CODE REVIEW

**Review by caroqliu — APPROVED** [review]
Is there anything special we need to to make sure the amp-img and amp-layout documentation updates on amp.dev given the file structure change? @sebastianbenz

**Comment by sebastianbenz** [comment]
We need to update the importer directory as well:

```
lib/pipeline/componentReferenceDocument.test.js
17:      githubPath: 'builtins/amp-test.md',

lib/pipeline/componentReferenceImporter.js
46:const BUILT_IN_PATH = 'builtins';
```

**Comment by rcebulko** [comment]
> We need to update the importer directory as well:
> 
> ```
> platform/lib/pipeline/componentReferenceDocument.test.js
> 17:      githubPath: 'builtins/amp-test.md',
> 
> platform/lib/pipeline/componentReferenceImporter.js
> 46:const BUILT_IN_PATH = 'builtins';
> ```

Done! Thanks for the catch, I had only been grepping for `/builtins` since there were so many other mentions of `builtins` in text/parts of other links.

@sebastianbenz @caroqliu How do you think is the best way to merge these two PRs? I'm not familiar with the AMP.dev release schedule, and I imagine timing it wrong would result in dead links for [insert duration of time here]

**Comment by caroqliu** [comment]
> > We need to update the importer directory as well:
> > ```
> > platform/lib/pipeline/componentReferenceDocument.test.js
> > 17:      githubPath: 'builtins/amp-test.md',
> > 
> > platform/lib/pipeline/componentReferenceImporter.js
> > 46:const BUILT_IN_PATH = 'builtins';
> > ```
> 
> Done! Thanks for the catch, I had only been grepping for `/builtins` since there were so many other mentions of `builtins` in text/parts of other links.
> 
> @sebastianbenz @caroqliu How do you think is the best way to merge these two PRs? I'm not familiar with the AMP.dev release schedule, and I imagine timing it wrong would result in dead links for [insert duration of time here]

Might be worthwhile to search `builtins/` too just in case.

re: timing, I believe the push to amp.dev is typically on Tuesday mornings. One thing we could do is merge the PR in AMPHTML followed by this one, and then do the push, all in somewhat quick succession (perhaps within 24 hours of the next release), to make sure the site is coordinated with the repo and vice versa. But I will defer to @sebastianbenz since he has more experience here.

**Comment by sebastianbenz** [comment]
amp.dev pulls in docs from the latest release branch. Once the builtins change is in prod, I'll merge this PR and the next amp.dev release will pull the docs from the correct branch.

**Comment by rcebulko** [comment]
@sebastianbenz github.com/ampproject/amphtml/pull/35139 has been merged 👍🏻 Thanks for the help with this!

**Comment by sebastianbenz** [comment]
Thanks for the update. Just verified that built import still works. This is good to merge once the built-ins  path change is in prod.

**Pull request merged**
